db/sg_replicate_cfg.go

```diff
@@ -93,6 +93,8 @@ type ReplicationConfig struct {
 	Cancel                 bool                      `json:"cancel,omitempty"`
 	Adhoc                  bool                      `json:"adhoc,omitempty"`
 	BatchSize              int                       `json:"batch_size,omitempty"`
+	// SGR1CheckpointID can be set to optionally fall back to when SGR2 checkpoints can't be found.
+	SGR1CheckpointID string `json:"sgr1_checkpoint_id,omitempty"`
```

Comment:
Do we actually want users to be able to specify this in their config/REST payloads?  Or is this just a convenient location to store this value to get it from the rest package down to db?  If it's just the latter, I think it muddies the intention of ReplicationConfig a bit, and opens up potential issues if people start setting the value.

It might be cleaner to send to sgrReplicateMgr.PutReplications in a separate map, or stick the new property on ReplicationCfg and modify sgReplicateMgr.PutReplications to accept that as input instead of ReplicationConfig.